File: mobile/ios/KinicApp/Services/VFSCandidLabels.swift
// Where: mobile/ios/KinicApp/Services/VFSCandidLabels.swift
// What: Candid record and variant label hashing for VFS-only calls.
// Why: Candid encodes field names as numeric hashes on the wire.

import Foundation

enum VFSCandidLabels {
    static func id(_ label: String) -> UInt32 {
        var hash: UInt32 = 0
        for byte in label.utf8 {
            hash = hash &* 223 &+ UInt32(byte)
        }
        return hash
    }
}

File: mobile/android/app/src/main/java/xyz/kinic/android/VfsCandidLabels.kt
// Where: mobile/android/app/src/main/java/xyz/kinic/android/VfsCandidLabels.kt
// What: Candid record and variant label hashing.
// Why: Candid encodes field names into numeric hashes on the wire.

package xyz.kinic.android

object VfsCandidLabels {
    fun id(label: String): UInt {
        var hash = 0u
        for (byte in label.encodeToByteArray()) {
            hash = hash * 223u + byte.toUByte().toUInt()
        }
        return hash
    }
}
